src/test/java/bio/terra/CloudStorageEventHarnessTest.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import bio.terra.cloudfunctions.common.GsonWrapper;
+import bio.terra.common.BaseTest;
+import com.google.events.cloud.storage.v1.StorageObjectData;
+import java.util.Map;
+import org.junit.Test;
+
+public class CloudStorageEventHarnessTest extends BaseTest {
+  @Test
+  public void acceptSupportedCloudEventTest() {
+    try {
+      GCSEventHarnessImpl instance = new GCSEventHarnessImpl();
+      Map<?, ?> m = GsonWrapper.convertFromClass(MOCK_GCS_EVENT_GZIP, Map.class);
+      instance.accept(m, GCS_CLOUD_EVENT_CONTEXT);
+      StorageObjectData event = instance.getEvent(StorageObjectData.class);
+      assertMockTGZStorageObjectData(event);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void rejectUnSupportedCloudEventTest() {
+    try {
+      GCSEventHarnessImpl instance = new GCSEventHarnessImpl();
+      Map<?, ?> m = GsonWrapper.convertFromClass(MOCK_GCS_EVENT_GZIP, Map.class);
+      instance.accept(m, UNSUPPORTED_CLOUD_EVENT_CONTEXT);
+      StorageObjectData event = instance.getEvent(StorageObjectData.class);
+      assertMockTGZStorageObjectData(event);
+    } catch (Exception e) {
+      assertEquals("Unexpected event type: unsupported.event", e.getMessage());
+    }
+  }
```

Comment:
When you're looking to verify that an exception is thrown, I think using `expected` as part of the `@Test` annotation is cleaner and more widely used.  As I mentioned elsewhere I think we can use java's built in IllegalArgumentException rather than our own class, but I've left that bit alone in my suggestion.

```suggestion
  @Test(expected = UnSupportedCloudEventTypeException.class)
  public void rejectUnSupportedCloudEventTest() throws Exception {
    GCSEventHarnessImpl instance = new GCSEventHarnessImpl();
    Map<?, ?> m = GsonWrapper.convertFromClass(MOCK_GCS_EVENT_GZIP, Map.class);
    instance.accept(m, UNSUPPORTED_CLOUD_EVENT_CONTEXT);
    StorageObjectData event = instance.getEvent(StorageObjectData.class);
    assertMockTGZStorageObjectData(event);
  }
```